service/raft/raft_rpc.cc

```diff
@@ -25,140 +25,109 @@ raft_ticker_type::time_point timeout() {
     return raft_ticker_type::clock::now() + raft_tick_interval * (raft::ELECTION_TIMEOUT.count() / 2);
 }
 
+void log_missing_raft_address(raft::server_id srv_id,
+        const std::experimental::source_location location = std::experimental::source_location::current()) {
```

Comment:
This `location` is constructed at the call site of the function, which currently is `one_way_rpc`/`two_way_rpc`. This is not desirable, becasue it means that the `location` will point inside `one_way_rpc`/`two_way_rpc`, while we want to know the concrete verb (e.g. `send_vote_request`).

So `one_way_rpc` and `two_way_rpc` functions should take a `location` parameter (default-initialized to `current()`) and pass it down to this function, and this function should not default-initialize `location`.

Reply:
Right, I first implemented this function and then managed to use templates. Do you know a better solution for determining the output type than decltype(verb())?

Reply:
https://en.cppreference.com/w/cpp/types/result_of

Reply:
Fixed.